fastplotlib/layouts/_base.py

```diff
@@ -224,12 +229,90 @@ def set_viewport_rect(self, *args):
         self.viewport.rect = self.get_rect()
 
     def render(self):
-        # does not flush
+        self._call_animate_functions(self._animate_funcs_pre)
+
+        # does not flush, flush must be implemented in user-facing Plot objects
         self.viewport.render(self.scene, self.camera)
 
         for child in self.children:
             child.render()
 
+        self._call_animate_functions(self._animate_funcs_post)
+
+    def _call_animate_functions(self, funcs: Iterable[callable]):
+        for fn in funcs:
+            try:
+                args = getfullargspec(fn).args
+
+                if len(args) > 0:
+                    if args[0] == "self" and not len(args) > 1:
+                        fn()
+                    else:
+                        fn(self)
+                else:
+                    fn()
+            except (ValueError, TypeError):
+                warn(
+                    f"Could not resolve argspec of {self.__class__.__name__} animation function: {fn}, "
+                    f"calling it without arguments."
+                )
+                fn()
+
+    def add_animations(
+        self,
+        *funcs: Iterable[callable],
+        pre_render: bool = True,
+        post_render: bool = False,
+    ):
+        """
+        Add function(s) that are called on every render cycle.
+        These are called at the Subplot level.
+
+        Parameters
+        ----------
+        *funcs: callable or iterable of callable
+            function(s) that are called on each render cycle
+
+        pre_render: bool, default ``True``, optional keyword-only argument
+            if true, these function(s) are called before a render cycle
+
+        post_render: bool, default ``False``, optional keyword-only argument
+            if true, these function(s) are called after a render cycle
+
+        """
+        for f in funcs:
+            if not callable(f):
+                raise TypeError(
+                    f"all positional arguments to add_animations() must be callable types, you have passed a: {type(f)}"
+                )
+            if pre_render:
+                self._animate_funcs_pre += funcs
+            if post_render:
+                self._animate_funcs_post += funcs
```

Comment:
edge case where user sets `pre_render` and `post_render` to true...would this call issues if the animation function would then be getting called twice?

Reply:
It wouldn't be an issue, it'd just get called twice. We could change to a kwarg that takes in a string of "pre" or "post"